homeassistant/components/tplink/switch.py

```diff
@@ -32,131 +34,51 @@ async def async_setup_entry(
     data: TPLinkData = hass.data[DOMAIN][config_entry.entry_id]
     parent_coordinator = data.parent_coordinator
     device = parent_coordinator.device
-    add_device_switch = True
-    for no_switch_module in NO_SWITCH_DEVICE_MODULES:
-        if no_switch_module in device.modules:
-            add_device_switch = False
-            break
-    entities: list = []
-    if add_device_switch and device.children:
-        # Historically we only add the children if the device is a strip
-        _LOGGER.debug("Initializing strip with %s sockets", len(device.children))
-        entities.extend(
-            SmartPlugSwitchChild(device, parent_coordinator, child)
-            for child in device.children
-        )
-    elif add_device_switch:
-        entities.append(SmartPlugSwitch(device, parent_coordinator))
 
-    if Module.Led in device.modules:
-        entities.append(
-            SmartPlugLedSwitch(device, parent_coordinator, device.modules[Module.Led])
-        )
+    entities = _entities_for_device_and_its_children(
+        device,
+        feature_type=Feature.Switch,
+        entity_class=Switch,
+        coordinator=parent_coordinator,
+    )
 
     async_add_entities(entities)
 
 
-class SmartPlugLedSwitch(CoordinatedTPLinkEntity, SwitchEntity):
-    """Representation of switch for the LED of a TPLink Smart Plug."""
-
-    _attr_translation_key = "led"
-    _attr_entity_category = EntityCategory.CONFIG
-
-    def __init__(
-        self, device: Device, coordinator: TPLinkDataUpdateCoordinator, led_module: Led
-    ) -> None:
-        """Initialize the LED switch."""
-        super().__init__(device, coordinator)
-        self._attr_unique_id = f"{device.mac}_led"
-        self._led_module = led_module
-        self._async_update_attrs()
-
-    @async_refresh_after
-    async def async_turn_on(self, **kwargs: Any) -> None:
-        """Turn the LED switch on."""
-        await self._led_module.set_led(True)
-
-    @async_refresh_after
-    async def async_turn_off(self, **kwargs: Any) -> None:
-        """Turn the LED switch off."""
-        await self._led_module.set_led(False)
-
-    @callback
-    def _async_update_attrs(self) -> None:
-        """Update the entity's attributes."""
-        self._attr_is_on = self._led_module.led
-
-    @callback
-    def _handle_coordinator_update(self) -> None:
-        """Handle updated data from the coordinator."""
-        self._async_update_attrs()
-        super()._handle_coordinator_update()
-
-
-class SmartPlugSwitch(CoordinatedTPLinkEntity, SwitchEntity):
-    """Representation of a TPLink Smart Plug switch."""
-
-    _attr_name: str | None = None
+class Switch(CoordinatedTPLinkEntity, SwitchEntity):
```

Comment:
```suggestion
class TPLinkSwitch(CoordinatedTPLinkEntity, SwitchEntity):
```
I think we agreed to stick with naming the platforms `TPLinkBlah` the other day.